tech-specs/src/explore/design/ibc.md

```diff
@@ -0,0 +1,408 @@
+                    //   - Get the stored commitment and compare it with a commitment made from the packet
+
+                    // Verify the proofs to check the acknowledgement has been written on the counterpart ledger
+                    //   - Use `ibc-rs::ics04_connection::handler::verify::verify_packet_acknowledgement_proofs()`
+                }
+
+                "commitments" => {
+                    let packet = get_packet(key, data_from_tx)?;
+                    // check if the commitment is deleted
+                    match check_commitment_state(key) {
+                        StateChange::Deleted => {
+                            // Assert that the packet was actually sent on this channel
+                            //   - Get the stored commitment and compare it with a commitment made from the packet
+
+                            // Check the channel state change
+                            match check_channel_state(channel_id) {
+                                StateChange::Closed => {
+                                    // "Timeout"
+                                    // Assert that the connection and channel are open
+
+                                    // Assert that the counterpart ledger has exceeded the timeout height or timestamp
+
+                                    // Assert that the packet sequence is the next sequence that the channel expects (Ordered channel)
+                                }
+                                _ => {
+                                    // "TimeoutOnClose"
+                                    // Assert that the packet sequence is the next sequence that the channel expects (Ordered channel)
+
+                                    // Verify the proofs to check the counterpart ledger's state is expected
+                                    //   - The channel state on the counterpart ledger should be CLOSED
+                                    //   - Use `ibc-rs::ics04_connection::handler::verify::verify_channel_proofs()`
+                                }
+                                // Verify the proofs to check the packet has not been confirmed on the counterpart ledger
+                                //   - For ordering channels, use `ibc-rs::ics04_connection::handler::verify::verify_next_sequence_recv()`
+                                //   - For not-ordering channels, use `ibc-rs::ics04_connection::handler::verify::verify_packet_receipt_absence()`
+                            }
+                        }
+                        StateChange::Created | StateChange::Updated => {
+                            // They are checked in other cases
+                        }
+                        _ => return Err(Error::InvalidStateChange("Invalid state change happened")),
+                    }
+                }
+
+                "ports" | "receipts" | "acks" => {
+                    // They are checked in other cases
```

Comment:
really? how do we assert that the correct writes are made in conjunction?

Reply:
Ah, they should be validated. Thanks!